test/jsonplaceholderconfig.py

```diff
@@ -59,6 +59,33 @@ class CreatePost(ResourceConfig):
     )
 
 
+class CreatePostWithSchema(ResourceConfig):
+
+    name = "create_post_with_schema"
+    path = ["posts"]
+    method = "POST"
+    description = "Create a new post"
+
+    schema = {
+                "title": "create post",
+                "description": "a schema to create a post",
+                "type": "object",
+                "properties": {
+                    "user": {
+                        "description": "id of the user",
+                        "type": "string"
+                    },
+                    "body": {
+                        "description": "the body of the post",
+                        "type": "string"
+                    }
+                },
+                "required": ["user", "body"]
+     }
+
+    post = BodyParameter(name=None, required=True, schema=schema)
```

Comment:
Currently a `BodyParameter` with `name=None` indicates a value that covers the whole body. In hindsight I'm not that fond of this approach. We expose an implementation detail to the user, namely that we implement whole-body support via class `BodyParameter`. I think that from a user's point-of-view, a new class, e.g. `Body`, would have been nicer. Alas, that's "voortschrijdend inzicht" - "new insights" - and not that relevant for this pull request.

Reply:
Agreed, `name=None` feels hacky. a `Body` class would be more elegant.